chromium_src/chrome/browser/ui/views/frame/browser_view.cc

```diff
@@ -7,12 +7,15 @@
 #include "brave/browser/ui/views/tabs/brave_tab_strip.h"
 #include "brave/browser/ui/views/toolbar/brave_toolbar_view.h"
 #include "chrome/browser/ui/views/frame/tab_strip_region_view.h"
+#include "chrome/browser/ui/views/incognito_clear_browsing_data_dialog.h"
 
 #define ToolbarView BraveToolbarView
 #define BrowserTabStripController BraveBrowserTabStripController
 #define TabStrip BraveTabStrip
+#define IncognitoClearBrowsingDataDialog BraveIncognitoClearBrowsingDataDialog
```

Comment:
`#define SetShowCloseButton(SHOW) SetShowCloseButton(false)`

Reply:
in incognito_clear_browsing_data_dialog.cc override

Reply:
make sure to include widget_delegate.h header in the chromium_src override before this define

Reply:
Cool! we can remove other override files then

Reply:
Sorry, I'm pretty much learning on the go here so I might need someone to explain this to me .....
Are we saying we will put  
`#define SetShowCloseButton(SHOW) SetShowCloseButton(false)`
into incognito_clear_browsing_data_dialog.cc file and that will override the default `SetShowCloseButton(true)` configuration? 🤔

Reply:
Yes, I think you got it right. As what you want to do with chromium_src overrides is only one thing, calling `SetShowCloseButton(false)`, defining the function-like macro can be more efficient way. We usually try to minimize override and patch files for sustainability. The macro will replace `SetShowCloseButton(true)` in original file to `SetShowCloseButton(false)` so we don't need other changes. And the reason you should include `widget_delegate.h` before that is we don't want to replace the function's declaration.

Reply:
Thanks @sangwoo108, at least I got the concept correct 💪 
Now, in term of execution, I got something that looks like this in the incognito_clear_browsing_data_dialog.cc 
```
#include "ui/views/widget/widget_delegate.h"

#define SetShowCloseButton(SHOW) SetShowCloseButton(false)

#include "src/chrome/browser/ui/views/incognito_clear_browsing_data_dialog.cc"
```
and it gave me errors while compiling 
![image](https://user-images.githubusercontent.com/31626127/184347708-a089b035-0d17-4f9d-bc88-f1f42c05133c.png)

I've been trying a few stuff but couldn't work it out, definitely feel like I'm lacking some knowledge, can anybody please drop me some hints? 🤔

Reply:
Aha, that would be solved by including "ui/views/window/dialog_delegate.h" before the function-like macro. At 405 line of dialog_delegate.h, there's a line `VIEW_BUILDER_PROPERTY(bool, ShowCloseButton)` and our macro thought it should be replaced but that's not we want.

As dialog_delegate.h includes widget_delegate.h, I think you can replace widget_delegate.h with dialog_delegate.h. Can you try this?


```
#include "ui/views/window/dialog_delegate.h"

#define SetShowCloseButton(SHOW) SetShowCloseButton(false)

#include "src/chrome/browser/ui/views/incognito_clear_browsing_data_dialog.cc"

Reply:
Ahhhh, thank you for the suggestion @sangwoo108
I tried to include `dialog_delegate.h` as you mentioned but it gave me another error (similar error) which related to `bubble_dialog_delegate_view.h` 
However, I found out that `bubble_dialog_delegate_view.h` also includes `dialog_delegate.h` so I ended up including "ui/views/bubble/bubble_dialog_delegate_view.h" into the file and it worked!!

Just a quick noob question on this since I'm kinda new to C++, I didn't quite understand how this macro replacing works,  I originally thought it could easily look for the targeted (used) SetShowCloseButton function, how could it "thought" that `VIEW_BUILDER_PROPERTY(bool, ShowCloseButton)` is equivalent to `SetShowCloseButton()`? 🤔 
I'll push my new changes shortly 💪

Reply:
> VIEW_BUILDER_PROPERTY(bool, ShowCloseButton) is equivalent to SetShowCloseButton()

Sorry, My previous comment made you confused. Let me try to explain what I understand again. 

When we have something defined with `#define` directive, the preprocessor will replace every same string with what we defined. Let's say we have a (hand waving) code 
```c++
#define PRINT_HELLO cout << "hello, world" << endl;

int main() {
  PRINT_HELLO  // << When preprocessor reach here, PRINT_HELLO will be replaced with code above. 
}
```

And in case of function-like macro, we should have argument for the macro.

```c++
#define PRINT(MESSGE) cout << MSG << endl;

int main() {
 PRINT("Hello, world') // When preprocessor reach here, it will replace this with an argument("hello, world"),

  PRINT // Err. When preprecessor reach here, it'll be like, "What is this? I need an argument(MSG) to replace PRINT"
}
```


Let's, get back to our code,
```c++
#define ShowCloseButton(SHOW) ShowCloseButton(false)

// the following line is inserted from  one of `#include`d files
VIEW_BUILDER_PROPERTY(bool, ShowCloseButton)  // Preprocessor finds "ShowCloseButton" and it's like, "Hey, I need an argument to ShowCloseButton replace it, but you didn't set it."
```

So, the error you encountered is saying that `ShowCloseButton` at the line needs an argument in order to be replaced with `ShowCloseButton(false)` successfully.
```diff
- VIEW_BUILDER_PROPERTY(bool, ShowCloseButton)            // preprocessor found `ShowCloseButton` but
+ VIEW_BUILDER_PROPERTY(bool, ShowCloseButton(false))   // failed to make it like this as it doens't have an argument.
```

Reply:
This is an excellent explanation, thank you 🙏 
It's exactly how I thought it'd behave, I guess I was just confused about the slight difference between `SetShowCloseButton` and `ShowCloseButton`, I'm assuming there's a part somewhere that let the macro knows these two are the same? 🤔

Reply:
> SetShowCloseButton and ShowCloseButton

Oh, sorry again. I missed that difference :( It's more complicated than I thought. VIEW_BUILD_PROPERTY is another macro to generate setter for property.
https://source.chromium.org/chromium/chromium/src/+/main:ui/views/metadata/view_factory.h;l=339?q=VIEW_BUILDER_PROPERTY&ss=chromium  . So I guess SetCloseButton was expanded from VIEW_BUILDER_PROPERTY first. But in this case, my previous explanation looks somewhat inaccurate.

Reply:
Ohhh mann, no worries at all, your example is awesomee 💯 
Really appreciate it 🙏